frontend/src/components/tile-providers.ts

```diff
@@ -0,0 +1,70 @@
+import * as L from "leaflet";
+import type { Option } from "@/components/AppSelect.vue";
+import "leaflet-providers";
+
+/** some pretty base layer tile providers */
+/** from https://leaflet-extras.github.io/leaflet-providers/preview/ */
+export const baseOptions = [
+  "Stadia.OSMBright",
+  "Stadia.AlidadeSmooth",
+  "Stadia.AlidadeSmoothDark",
+  "Stadia.StamenTerrainBackground",
+  "Stadia.StamenTonerLite",
+  "Stadia.StamenToner",
+  "Stadia.StamenTerrain",
+  "Stadia.Outdoors",
+  "Stadia.StamenWatercolor",
+  "CartoDB.Voyager",
+  "CartoDB.VoyagerNoLabels",
+  "CartoDB.VoyagerLabelsUnder",
+  "CartoDB.Positron",
+  "CartoDB.PositronNoLabels",
+  "CartoDB.PositronOnlyLabels",
+  "OpenStreetMap.Mapnik",
+  "OpenStreetMap.HOT",
+  "Esri.WorldTopoMap",
+  "Esri.WorldTerrain",
+  "Esri.WorldShadedRelief",
+  "Esri.WorldPhysical",
+  "Esri.WorldImagery",
+  "Esri.WorldGrayCanvas",
+  "Esri.WorldStreetMap",
+  "Esri.DeLorme",
+  "USGS.USTopo",
+  "USGS.USImagery",
+  "USGS.USImageryTopo",
+  "OpenTopoMap",
+  "NASAGIBS.ViirsEarthAtNight2012",
+  "Stadia.StamenTerrainLines",
+  "Stadia.StamenTonerLines",
+  "Stadia.StamenTonerLabels",
+  "Stadia.StamenTerrainLabels",
+  "OpenRailwayMap",
+].map(nameToOption) satisfies Option[];
+
+/** get selectable option from tile provider name */
+function nameToOption(name: string) {
```

Comment:
Turns a list of provider names into human readable labels and thumbnail images. Worth looking at this code a bit.

Reply:
Sounds good; I think waiting for a use to change it makes sense.